src/saltext/vmware/utils/common.py

```diff
@@ -729,17 +783,19 @@ def get_vm_datacenter(*, vm):
     return datacenter
 
 
-def get_mors_type(obj, type):
+def get_parent_of_type(mors, type):
     """
-    Return a vim type from managed object reference
+    Finds the first parent of a managed object that matches the type specified.
+
+    `None` is returned if no object is found.
     """
-    datacenter = None
+    result = None
     while True:
-        if isinstance(obj, type):
-            datacenter = obj
+        if isinstance(mors, type):
+            result = mors
             break
         try:
-            obj = obj.parent
+            mors = mors.parent
         except AttributeError:
             break
-    return datacenter
+    return result
```

Comment:
**suggestion (non-blocking)** this *could* take advantage of early returns, and be simplified with:

```
while True:
    if isinstance(mors, type):
        return mors
    try:
        mors = mors.parent
    except AttributeError:
        return None
```

typically I'm not a fan of early returns, but it actually looks better here :thinking:

Reply:
I agree.